src/components/tl/mlx5/tl_mlx5_team.c

```diff
@@ -132,9 +134,21 @@ ucc_status_t ucc_tl_mlx5_team_create_test(ucc_base_team_t *team)
             ucc_tl_mlx5_team_destroy(team);
             return tl_team->status[1];
         }
+        tl_team->state = TL_MLX5_TEAM_STATE_A2A_INIT;
+    case TL_MLX5_TEAM_STATE_A2A_INIT:
+        status = ucc_tl_mlx5_a2a_init_start(tl_team);
+        if (status != UCC_OK) {
+            tl_error(UCC_TL_TEAM_LIB(tl_team),
+                     "failed to init a2a: %s",
+                     ucc_status_string(status));
+            return status;
+        }
+        tl_team->state = TL_MLX5_TEAM_STATE_A2A_POSTED;
+    case TL_MLX5_TEAM_STATE_A2A_POSTED:
+        status = ucc_tl_mlx5_a2a_init_progress(tl_team);
```

Comment:
if there is an error in init_progress you need to free a2a->net.dcis which you did malloc for in init_start

Reply:
right, I moved dcis this to a2a_init_progress and handle the error there